graph-builder/src/registry.rs

```
@@ -37,24 +46,163 @@ impl Into<cincinnati::Release> for Release {
     }
 }
 
+fn trim_protocol(src: &str) -> &str {
+    src.trim_left_matches("https://")
+        .trim_left_matches("http://")
+}
+
+fn get_credentials_or_empty_reader(credentials_path: &Option<PathBuf>) -> Box<dyn std::io::Read> {
+    let empty = Box::new(std::io::empty());
+    match credentials_path {
+        Some(path) => match File::open(path) {
+            Ok(file) => Box::new(file),
+            _ => empty,
```

We shouldn't eat this error.